File: backend/src/main/java/com/unisync/service/BookingService.java
package com.unisync.service;

import com.unisync.dto.BookingRequestDTO;
import com.unisync.dto.BookingResponseDTO;
import com.unisync.enums.BookingStatus;

import java.util.List;

public interface BookingService {
    BookingResponseDTO createBooking(BookingRequestDTO request, String userId);
    List<BookingResponseDTO> getUserBookings(String userId);
    List<BookingResponseDTO> getAllBookings();
    BookingResponseDTO updateBookingStatus(String bookingId, BookingStatus status, String rejectionReason);
    BookingResponseDTO cancelBooking(String bookingId, String userId);
    BookingResponseDTO getBookingById(String bookingId);
    void deleteBooking(String bookingId);
}

File: frontend/src/api/bookingService.js
import api from './axiosConfig';

const bookingService = {
  createBooking: async (bookingData) => {
    const response = await api.post('/bookings', bookingData);
    return response.data;
  },

  getMyBookings: async () => {
    const response = await api.get('/bookings/me');
    return response.data;
  },

  getAllBookings: async () => {
    const response = await api.get('/bookings');
    return response.data;
  },

  getBookingById: async (id) => {
    const response = await api.get(`/bookings/${id}`);
    return response.data;
  },

  updateBookingStatus: async (id, status, rejectionReason = null) => {
    const response = await api.patch(`/bookings/${id}/status`, {
      status,
      rejectionReason
    });
    return response.data;
  },

  cancelBooking: async (id) => {
    const response = await api.patch(`/bookings/${id}/status`, {
      status: 'CANCELLED'
    });
    return response.data;
  }
};

export default bookingService;
